### PR TITLE
VAULT-12144: add openapi responses for /sys/tools endpoints

### PR DESCRIPTION
**Pull request by dhuckins**
|Path|Operation|response|docs|
|----|----|----|----|
|/sys/tools/hash|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/vault/logical_system.go#L4081-L4085)||
|/sys/tools/hash/{urlalgorithm}|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/vault/logical_system.go#L4081-L4085)||
|/sys/tools/random|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/helper/random/random_api.go#L98-L102)||
|/sys/tools/random/{source}|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/helper/random/random_api.go#L98-L102)||
|/sys/tools/random/{source}/{urlbytes}|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/helper/random/random_api.go#L98-L102)||
|/sys/tools/random/{urlbytes}|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/9eaf2671634207d8ecacc69860315f0731fd686a/helper/random/random_api.go#L98-L102)||

added response structure tests

### CODE REVIEW

File: vault/logical_system_paths.go [hunk 1]
```diff
@@ -937,8 +951,21 @@ func (b *SystemBackend) toolsPaths() []*framework.Path {
...
-			Callbacks: map[logical.Operation]framework.OperationFunc{
-				logical.UpdateOperation: b.pathRandomWrite,
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: b.pathRandomWrite,
+					Responses: map[int][]framework.Response{
+						http.StatusOK: {{
+							Description: "OK",
+							Fields: map[string]*framework.FieldSchema{
+								"random_bytes": {
```
Comment (AnPucel) [inline vault/logical_system_paths.go:961]: WRT another PR comment, looks like this callback also can return warnings, should we include those here?

File: vault/logical_system_test.go [hunk 2]
```diff
@@ -3290,12 +3306,19 @@ func TestSystemBackend_ToolsHash(t *testing.T) {
...
 	req := logical.TestRequest(t, logical.UpdateOperation, "tools/random")
 
-	_, err := b.HandleRequest(namespace.RootContext(nil), req)
+	resp, err := b.HandleRequest(namespace.RootContext(nil), req)
 	if err != nil {
 		t.Fatalf("err: %v", err)
 	}
+	schema.ValidateResponse(
+		t,
+		schema.FindResponseSchema(t, paths, 1, req.Operation),
```
Comment (averche) [inline vault/logical_system_test.go:3318]: If it's not too much effort, could you please convert these to `GetResponseSchema(t, b.Route(...), ...)`

Comment (dhuckins) [reply vault/logical_system_test.go:3318]: will do! thought i got em all

File: vault/logical_system_test.go [hunk 3]
```diff
@@ -3174,14 +3174,21 @@ func TestSystemBackend_PluginCatalog_CannotRegisterBuiltinPlugins(t *testing.T)
...
 		"input": "dGhlIHF1aWNrIGJyb3duIGZveA==",
 	}
-	_, err := b.HandleRequest(namespace.RootContext(nil), req)
+	resp, err := b.HandleRequest(namespace.RootContext(nil), req)
 	if err != nil {
 		t.Fatalf("err: %v", err)
 	}
+	schema.ValidateResponse(
+		t,
+		schema.FindResponseSchema(t, paths, 0, req.Operation),
```
Comment (dhuckins) [inline vault/logical_system_test.go:3188]: forgot to update this one

**Review by averche — APPROVED** [review]

**Pull request merged**